modules/interfaces/src/custom-command-api-definition.ts

```diff
@@ -1,32 +1,43 @@
 import { GeneralCustomCommand } from "./command";
 import { GeneralCustomCommandResult } from "./command-result";
 import { GeneralCustomCommandRequestData } from "./request-data";
-import { GeneralDefinition } from "./entity-definition";
+import {
+  RestApiDefinition,
+  GeneralRestApiSettings
+} from "./entity-rest-api-definition";
 import { Session } from "./session";
 
-export interface CustomCommandDefinition extends GeneralDefinition {
+export interface CustomCommandApiDefinition extends RestApiDefinition {
   authorize?(
     reqData: GeneralCustomCommandRequestData,
-    session?: Session
+    session: Session | undefined,
+    settings: GeneralRestApiSettings
```

Comment:
I understand what you mean to point out. There are many cases that `settings` is unnecessary in the `authorize()` method.
However, we should offer its type as required one because `settings` must exist when it is needed.

It is okay to define a class like this.

```ts
class FooCommandApiDefinition extends CustomCommandApiDefinition {
  async authorize(reqData: GeneralCustomCommandRequestData) {
     ...
  }
}
```

Even when `authorize()` method only takes one argument, it's regarded as a valid extender. This is called contravariant characteristics of function arguments.

Reply:
What makes you confused maybe that `settings` is not changed by request, while `reqData` and `session` are. They might be passed via an alternative way.

##  Curried function
```ts
const FooDef = (settings) => class implements RestApiDefinition {
  async authorize(reqData, session) {
  }
}
```

## "this" keyword
```ts
class implements RestApiDefinition {
  async authorize(reqData, session) {
    this.settings // GeneralRestApiSettings
  }
}
```

I'm considering whether the first one can be the alternative, although I've never seen libraries that require developers to implement curried classes.